File: 2015/Day_02/Ruby/test_part_2.rb
require_relative "part_2"
require 'test/unit'

class TestWrappingPaper < Test::Unit::TestCase
    
    def test_get_dimensions
        assert_equal([2, 3, 4], get_dimensions('2x3x4'))
    end
    
    def test_calculate_bow_length
        assert_equal(24, calculate_bow_length([4,3,2]))
        assert_equal(10, calculate_bow_length([1,1,10]))
    end
    
    def test_calculate_ribbon_length
        assert_equal(10, calculate_ribbon_length([4,3,2]))
        assert_equal(4, calculate_ribbon_length([1,1,10]))
    end
end

File: 2015/Day_02/Python/test_part_2.py
from . import part_2


def test_get_dimensions():
    assert part_2.get_dimensions("2x3x4") == [2, 3, 4]
    assert part_2.get_dimensions("1x1x10") == [1, 1, 10]


def test_determine_lengths():
    assert part_2.determine_lengths([2, 3, 4]) == (10, 24)
    assert part_2.determine_lengths([1, 1, 10]) == (4, 10)
